components/AppShell.vue

```diff
@@ -0,0 +1,176 @@
+<template>
+  <div class="min-h-full">
+    <TransitionRoot
+      as="template"
+      :show="sidebarOpen"
+    >
+      <Dialog
+        as="div"
+        class="relative z-40 md:hidden"
+        @close="sidebarOpen = false"
+      >
+        <TransitionChild
+          as="template"
+          enter="transition-opacity ease-linear duration-300"
+          enter-from="opacity-0"
+          enter-to="opacity-100"
+          leave="transition-opacity ease-linear duration-300"
+          leave-from="opacity-100"
+          leave-to="opacity-0"
+        >
+          <div class="fixed inset-0 bg-gray-600 bg-opacity-75" />
+        </TransitionChild>
+
+        <div class="fixed inset-0 z-40 flex">
+          <TransitionChild
+            as="template"
+            enter="transition ease-in-out duration-300 transform"
+            enter-from="-translate-x-full"
+            enter-to="translate-x-0"
+            leave="transition ease-in-out duration-300 transform"
+            leave-from="translate-x-0"
+            leave-to="-translate-x-full"
+          >
+            <DialogPanel class="relative flex w-full max-w-xs flex-1 flex-col bg-surface-primary pt-5 pb-4">
+              <TransitionChild
+                as="template"
+                enter="ease-in-out duration-300"
+                enter-from="opacity-0"
+                enter-to="opacity-100"
+                leave="ease-in-out duration-300"
+                leave-from="opacity-100"
+                leave-to="opacity-0"
+              >
+                <div class="absolute top-0 right-0 -mr-12 pt-2">
+                  <!-- FIXME: This should be a button -->
```

Comment:
There are two instances of components that _should_ be buttons but our styling of the button html tag is affecting them. Removing the styling of a button can be done in a separate task and then this can be fixed.